adapters/rest_generic/rest_generic_connect.php

```diff
@@ -267,7 +277,12 @@ public function do_connect() {
 
 			// extract token
 			if ($this->rest_json) {
-				$this->key = (string)($this->json_path->find($result, $this->token_jsonpath)[0]);
+				if($this->auth_mode == ""){
```

Comment:
Could be `if (empty($this->auth_mode)) {`

Reply:
@ubihgu 
what is the status of this PR ?